### PR TITLE
Add verification if variable is array

### PR DESCRIPTION
**Pull request by henricavalcante**

### CODE REVIEW

**Comment by divanvisagie** [comment]
Side note: Looks like sum doesn't have any tests https://github.com/rwaldron/johnny-five/blob/master/test/fn.js

File: lib/fn.js [hunk 1]
```diff
@@ -105,11 +105,17 @@ Fn.square = function(x) {
...
 Fn.sum = function sum(values) {
-  return values.reduce(function(accum, value) {
-    return accum + value;
-  }, 0);
+  if (values instanceof Array) {
+    return values.reduce(function(accum, value) {
+      return accum + value;
+    }, 0);
+  }
+  return;
```
Comment (divanvisagie) [inline lib/fn.js]: @rwaldron Should we be returning undefined here? maybe 0 instead?

Comment (rwaldron) [inline lib/fn.js]: I'd rather this be: 

```js
var vals;

if (Array.isArray(values)) {
  vals = values;
} else {
  vals = [].slice.call(arguments);
}

return vals.reduce(...);
```

Something like that...

**Comment by divanvisagie** [comment]
@henricavalcante Would you mind making the changes @rwaldron mentioned here? Also If I may ask , could you add tests to fn for sum that cover both it's current functionality and the test instance you are trying to guard against?

**Comment by henricavalcante** [comment]
I'm think that Travis is kidding me:

```
henrimichel@machine:~/git/fork/johnny-five (master)$ npm test

> johnny-five@0.8.38 test /Users/henrimichel/GIT/fork/johnny-five
> grunt

Running "jshint:files" (jshint) task
>> 232 files lint free.

Running "jscs:files" (jscs) task
>> 232 files without code style errors.

Running "nodeunit:tests" (nodeunit) task
Testing board.js.....................OK
Testing board-connection.js.OK
Testing compass.js........OK
Testing options.js.OK
Testing board.pins.js.......OK
Testing board.component.js..........OK
Testing capabilities.js.......OK
Testing accelerometer.js.......OK
Testing animation.js..........OK
Testing button.js........OK
Testing distance.js.......OK
Testing esc.js..............OK
Testing fn.js.........OK
Testing gyro.js......OK
Testing imu.js....OK
Testing lcd.js.............OK
Testing led.js........................................OK
Testing ledcontrol.js.............................................OK
Testing motor.js......................................................OK
Testing pin.js....................OK
Testing piezo.js..................OK
Testing ping.js....OK
Testing pir.js.....OK
Testing reflectancearray.js...........OK
Testing relay.js.......OK
>> .OK
Testing sensor.js................OK
Testing servo.js..........................OK
Testing shiftregister.js..OK
Testing sonar.js........OK
Testing stepper.js.....OK
Testing temperature.js...........OK
Testing switch.js...OK
Testing wii.js........OK
>> 1916 assertions passed (2704ms)

Done, without errors.
henrimichel@machine:~/git/fork/johnny-five (master)$ 
```

**Comment by rwaldron** [comment]
Landed https://github.com/rwaldron/johnny-five/commit/3c4fbe60d883eb15543b13ea0cb2a8667de84014

(cherry picked the relevant fix to `Fn.sum`, intentionally omitted the style changes)

**Pull request closed without merging**